src/server.lisp

```diff
@@ -202,18 +203,23 @@ DISPATCH-TABLE and LOGGING-STREAM are both required arguments.  TIMEOUT is of ty
                              (unless f
                                (error 'unknown-rpc-method
                                       :method-name (|RPCRequest-method| request)))
-                             (setf result
-                                   (if timeout
-                                       (bt:with-timeout (timeout)
-                                         (apply f (concatenate 'list args-as-list kwargs-as-plist)))
-                                       (apply f (concatenate 'list args-as-list kwargs-as-plist))))
-
+                             (flet ((apply-handler ()
+                                      (handler-bind
+                                          ((error (lambda (c)
+                                                    (when debug
+                                                      (cl-syslog:format-log logger ':err (trivial-backtrace:print-backtrace c :output nil))))))
+                                        (apply f (concatenate 'list args-as-list kwargs-as-plist)))))
+                               (setf result
+                                     (if timeout
+                                         (bt:with-timeout (timeout)
+                                           (apply-handler))
+                                         (apply-handler))))
```

Comment:
I don't feel great about adding yet more levels of nesting/indentation here... if no one objects I might follow this up with a PR to break `%rpc-server-thread-worker` into a series of smaller functions to aid readability.

The question [remains](https://github.com/rigetti/rpcq/pull/95#discussion_r344363912) as to whether the intent of #87 is to merely log backtraces, or add them to the `|RPCError|` reply. Or perhaps invoke the debugger instead.

Reply:
I wouldn't object to improving readability here.